### PR TITLE
Add servlet that returns all the game information on the main page

### PR DESCRIPTION
**Pull request by ldcWV**
http://ldchen-step-2020.appspot.com/

### CODE REVIEW

File: src/main/java/com/google/sps/data/BasicGameData.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* Represents only the info that is needed about a game from the main page.
+*/
+public class BasicGameData {
+    //TODO(ldchen): delete all of the mocking once datastore is working.
+    private int getRandomIntegerBetween(int l, int r) {
+        int res = l + (int)(Math.random()*(r-l+1));
+        if(res > r) res = r; // happens if Math.random() somehow gives exactly 1
+        return res;
+    }
+    private String getRandomWord() {
```
Comment (saramissak) [inline src/main/java/com/google/sps/data/BasicGameData.java:31]: add extra line here between last function and this function

File: src/main/java/com/google/sps/data/Coordinates.java [hunk 2]
```diff
@@ -70,4 +72,34 @@ public double getLatitude() {
...
+        return new Coordinates(-90+Math.random()*180, -180+Math.random()*360);
+    }
+
+    /**
+    * Get random coordinates whose latitude and longitude values are at most d away from this.
+    * If these coordinates are (x,y), then this will return a point whose x is between x-d and x+d,
+    * and whose y is between y-d and y+d.
+    * @param d the upper bound on the difference between the coordinates.
+    */
+    public Coordinates getRandomWithin(double d) {
```
Comment (saramissak) [inline src/main/java/com/google/sps/data/Coordinates.java]: Please be more specific for what 'd' is, maybe  "differenceForUpperBound"?

File: src/main/java/com/google/sps/data/MockDatastoreManager.java [hunk 3]
```diff
@@ -129,4 +141,40 @@ private Hint getHint(int idx) {
...
         return res.build();
     }
+
+    /**
+    * Returns a random integer in the interval [l,r].
+    * @param l the left endpoint of the interval (inclusive).
+    * @param r the right endpoint of the interval (inclusive).
+    * @return a random integer in [l,r].
+    */
+    private int getRandomIntegerBetween(int l, int r) {
```
Comment (saramissak) [inline src/main/java/com/google/sps/data/MockDatastoreManager.java]: 'l' and 'r' are unclear. Maybe lowerBound and upperBound are clearer variable names.

File: src/main/java/com/google/sps/data/MockDatastoreManager.java [hunk 4]
```diff
@@ -129,4 +141,40 @@ private Hint getHint(int idx) {
...
+    }
+
+    /**
+    * Creates a game with a random id, title, difficulty, and stars.
+    * All other fields are left as default.
+    * @return a random Game.
+    */
+    private Game getRandomGame() {
+        Game.Builder res = new Game.Builder(IDGenerator.gen(), getRandomWord() + " " + getRandomWord());
+        int numStarVotes = getRandomIntegerBetween(0, 10);
```
Comment (saramissak) [inline src/main/java/com/google/sps/data/MockDatastoreManager.java:172]: Isn't the star scale 1-5? Why is there 10 here?

File: src/main/java/com/google/sps/data/MockDatastoreManager.java [hunk 5]
```diff
@@ -129,4 +141,40 @@ private Hint getHint(int idx) {
...
+        res.setNumStarVotes(numStarVotes);
+        res.setTotalStars(getRandomIntegerBetween(numStarVotes, 5*numStarVotes));
+        int numDifficultyVotes = getRandomIntegerBetween(0, 10);
+        res.setNumDifficultyVotes(numDifficultyVotes);
+        res.setTotalDifficulty(getRandomIntegerBetween(0, 2*numDifficultyVotes));
+        return res.build();
+    }
 }
```
Comment (saramissak) [inline src/main/java/com/google/sps/data/MockDatastoreManager.java]: Add extra line at the end of file.

File: src/main/java/com/google/sps/data/BasicGameData.java (hunk 1 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/data/BasicGameData.java:31]: Done, thanks.

File: src/main/java/com/google/sps/data/Coordinates.java (hunk 2 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/data/Coordinates.java]: Thanks, renamed to "maxDelta".

File: src/main/java/com/google/sps/data/MockDatastoreManager.java (hunk 3 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/data/MockDatastoreManager.java]: Thanks, renamed to "left" and "right" which is what they're supposed to mean.

File: src/main/java/com/google/sps/data/MockDatastoreManager.java (hunk 4 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/data/MockDatastoreManager.java:172]: numStarVotes is the number of people who voted, not the number of stars.

File: src/main/java/com/google/sps/data/MockDatastoreManager.java (hunk 5 above)
Comment (ldcWV) [reply src/main/java/com/google/sps/data/MockDatastoreManager.java]: Done, thanks.

**Pull request merged**